ConnectionFactory.php

```diff
@@ -226,19 +239,15 @@ private function addDatabaseSuffix(array $params): array
      *                 URL extracted into individual parameter parts.
      * @psalm-return Params
      *
-     * @throws Exception
+     * @throws DBALException
      */
     private function parseDatabaseUrl(array $params): array
     {
         if (! isset($params['url'])) {
             return $params;
         }
 
-        try {
-            $parsedParams = $this->dsnParser->parse($params['url']);
-        } catch (MalformedDsnException $e) {
-            throw new Exception('Malformed parameter "url".', 0, $e);
```

Comment:
I removed this since `MalformedDsnException` is an instance of `Doctrine\DBAL\Exception`

Reply:
IIRC, this was added to have a better error message for the user

Reply:
ok I see. Then we need to check how we can have more specific exception as we cannot instantiate an arbitrary `Doctrine\DBAL\Exception` anymore on DBAL 4.